posts/napari-cosmx-basics/using-napari-cosmx.qmd

```diff
@@ -0,0 +1,501 @@
+dna.contrast_limits = [507.82222222222225, 1260.8000000000002]
+dna.gamma = 1.1682758620689655
+```
+
+Screenshots can be done programmatically with the napari's `screenshot` method
+and there are additional settings you can change (_e.g._, just the canvas, scale)
+that we won't cover here.
+
+```{r}
+#| eval: false
+fig_path = output_path + "/fig-transcripts.png"
+with imageio.get_writer(fig_path, dpi=(800, 800)) as writer:
+        screenshot = gem.viewer.screenshot(canvas_only=False)
+        writer.append_data(screenshot)
+```
+
+There are also methods available in `napari-cosmx` that do not have the GUI
+equivalent. We won't be able to touch on all of these methods in this post but I 
+want to highlight a few. 
+
+## Plot transcripts with an expanded color pallette
+
+The GUI offers a handful of colors to plot transcripts. We can specify which color,
+by name or by hexcode, to plot. For example: 
+
+```{r}
+#| eval: false
+gem.plot_transcripts(gene = "Calb1", color = 'pink', point_size=20)
+```
+
+which is the same as
+
+```{r}
+#| eval: false
+gem.plot_transcripts(gene = "Calb1", color = '#FFC0CB', point_size=20)
+```
+
+## Plotting genes with list comprehensions
+
+We can plot similar genes or targets with the same color. For example, here's
+the negatives in white.
+
+```{r}
+#| eval: false
+#| 
+df = gem.targets
+filtered_df = df[df.target.str.contains("NegPrb")]
+
+pandas_df = filtered_df.to_pandas_df()
+negatives = pandas_df.target.unique().tolist()
+[gem.plot_transcripts(gene = x, color = "white", point_size=10) for x in negatives];
+
+
+```
+
+We can also combine targets with a list of tuples:
+
+```{r}
+#| eval: false
+genes = [('Npy', "orange"), ("Calb1", "cyan")]
+[gem.plot_transcripts(gene = x[0], color = x[1], point_size=10) for x in genes];
+
+```
+
+## Changing transcript transparency
+
+Sometimes transcripts can be stacked on top of each other to the point you cannot
+discern any features. For example (to do). Adjusting the transparents of the layer
+in the GUI only changes the transparency of a single point so that's not feasible.
+But this is possible in interepeter. 
+
+```{r}
+#| eval: false
+gem.viewer.layers['Npy'].opacity = 0.2
+
+```
+
+## Plot all transcripts
+
+This is not advised for resource-limited systems as it plots all transcripts. 
+The method `add_points` plots all the points for a given FOV. If no FOV is
+specified, it will plot all transcripts (this can be taxing on resource-limited
+computers).
+
+```{r}
+#| eval: false
+
+gem.add_points(fov=1)
+```
+
+## Specify individual cell types
+
+You can choose the color of the cell types and which cells to color by supplying
+a dictionary to the 
+
+```{r}
+#| eval: false
+custom_colors = {
+"EPEN":"#AA0DFE",
+"CPEC":"#3283FE",
+"GN":"#85660D",
+"VLMC":"#1C8356"
+}
+
+gem.color_cells('RNA_nbclust_clusters', color=custom_colors)
+
+```
+
+## Color cells with outlines
+
+We can also plot the cell colors as boundaries instead of filled in polygons (@fig-contours).
```

Comment:
Would be good to adjust the camera center and zoom to avoid FOV borders in the figure.

Reply:
done.